src/test/java/StringManipulationTest.java

```diff
@@ -19,131 +22,206 @@ public void tearDown() {
     @Test
     public void testCount4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Hello, World!");
+        int count = manipulatedstring.count();
+        assertEquals(2, count);
     }
 
+    // DO NOT EDIT
     @Test
     public void testRemoveNthCharacter1() {
         manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
         assertEquals("I' bttr uts0e 16tsinths trn6 rgh?", manipulatedstring.removeNthCharacter(3, false));
     }
 
+    // DO NOT EDIT
     @Test
     public void testRemoveNthCharacter2() {
         manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
-        assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(3, true));
+        assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?",
+                manipulatedstring.removeNthCharacter(3, true));
     }
 
+    // Test case to verify removing nth character in a different string
     @Test
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I'm learning to code!");
+        assertEquals("I'm lea nin  to cod !", manipulatedstring.removeNthCharacter(4, true));
     }
 
+    // Test case for removing nth character with an out-of-bounds value of n
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Test string");
+        assertThrows(IndexOutOfBoundsException.class, () -> manipulatedstring.removeNthCharacter(13, true));
     }
 
+    // Test case to verify removing nth character without maintaining spacing in a
+    // different string
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is a test.");
+        assertEquals("Ths s  tst", manipulatedstring.removeNthCharacter(3, false));
     }
 
+    // Test case for removing nth character with an invalid value of n
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Test string");
+        assertThrows(IllegalArgumentException.class, () -> manipulatedstring.removeNthCharacter(0, true));
     }
 
+    // Test case to verify removing nth character in a long string
     @Test
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I am testing the removeNthCharacter method");
+        assertEquals("I a   e t n   h   e o e t C a a t r m t o ", manipulatedstring.removeNthCharacter(2, true));
+    }
+
+    // Test case for removing nth character from a null string
+    @Test
+    public void testRemoveNthCharacter8() {
+        manipulatedstring.setString(null);
+        assertThrows(NullPointerException.class, () -> manipulatedstring.removeNthCharacter(2, true));
     }
 
```

Comment:
testing an empty string to make sure there are no unintentional exceptions thrown could be useful

Reply:
added test case and functionality.